src/network/networkd-route.c

```diff
@@ -516,6 +518,61 @@ int route_expire_handler(sd_event_source *s, uint64_t usec, void *userdata) {
         return 1;
 }
 
+static int append_nexthop_one(Route *route, MultipathRoute *m, struct rtattr *rta, size_t max_length, struct rtnexthop *rtnh) {
+        int r;
+
+        rtnh->rtnh_len = sizeof(*rtnh);
+        rta->rta_len += rtnh->rtnh_len;
+
+        if (route->family == m->gateway.family) {
+                r = rtattr_append_attribute(rta, max_length, RTA_GATEWAY, &m->gateway.address, FAMILY_ADDRESS_SIZE(m->gateway.family));
+                if (r < 0)
+                        return r;
+                rtnh->rtnh_len += RTA_SPACE(FAMILY_ADDRESS_SIZE(m->gateway.family));
+        } else {
+                r = rtattr_append_attribute(rta, max_length, RTA_VIA, &m->gateway, FAMILY_ADDRESS_SIZE(m->gateway.family) + sizeof(m->gateway.family));
+                if (r < 0)
+                        return r;
+                rtnh->rtnh_len += RTA_SPACE(FAMILY_ADDRESS_SIZE(m->gateway.family) + sizeof(m->gateway.family));
+        }
+
+        rtnh->rtnh_ifindex = m->ifindex;
+        if (m->weight > 0)
+                rtnh->rtnh_hops = m->weight - 1;
+
+        return 0;
+}
+
+static int append_nexthops(Route *route, sd_netlink_message *req) {
+        char buf[4096] = {};
+        struct rtattr *rta = (struct rtattr *) buf;
+        struct rtnexthop *rtnh;
+        MultipathRoute *m;
+        Iterator i;
+        int r;
+
+        if (ordered_set_isempty(route->multipath_routes))
+                return 0;
+
+        rta->rta_type = RTA_MULTIPATH;
+        rta->rta_len = RTA_LENGTH(0);
+        rtnh = RTA_DATA(rta);
+
+        ORDERED_SET_FOREACH(m, route->multipath_routes, i) {
+                r = append_nexthop_one(route, m, rta, 4096, rtnh);
+                if (r < 0)
```

Comment:
May be add a debug log here.

Reply:
The caller of this function logs about errors. So, I do not think it is necessary to add logs here.